pkg/cmd/cli/cmd/login/loginoptions.go

```diff
@@ -121,7 +121,7 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 		// certificate authority unknown, check or prompt if we want an insecure
 		// connection or if we already have a cluster stanza that tells us to
 		// connect to this particular server insecurely
-		case x509.UnknownAuthorityError:
+		case x509.UnknownAuthorityError, x509.HostnameError, x509.CertificateInvalidError:
 			if o.InsecureTLS ||
 				hasExistingInsecureCluster(*clientConfig, *o.StartingKubeConfig) ||
 				promptForInsecureTLS(o.Reader, o.Out) {
```

Comment:
@liggitt Sounds good, should we also show the original error message when prompting (so that in case of `HostnameError` it displays the expected/actual hostnames), or would a generic message be enough?